tools/juliapkg/src/config.jl

```diff
@@ -3,6 +3,7 @@ Configuration object
 """
 mutable struct Config
     handle::duckdb_config
+    registered_objects::Dict{Any, Any}
```

Comment:
This is unused still, I'm not sure yet if we need this or not, to safeguard objects passed to the custom buffer manager from being garbage collected?

Reply:
I just merged with master, all the conflicts were fortunately very easy to resolve.
As a bonus, I was re-running the tests to check if everything still worked, and I also tested to see if the bug I identified earlier ^ was still there.
It seems to be fixed! Either by the other changes to the buffer manager or by the changes we made to Julia (though I also encountered the bug in the C-api, so the Julia angle seems unlikely)